### PR TITLE
Implementation of the home page

### PR DESCRIPTION
**Pull request by briansetzesglobant**
The HomePage screen was implemented

https://user-images.githubusercontent.com/84338449/204357195-8788f466-06f8-4154-97b2-47ef1b9d483f.mp4

### CODE REVIEW

File: lib/src/core/resource/data_state.dart [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+class DataEmpty<T> extends DataState<T> {
+  const DataEmpty()
+      : super(
+          type: DataStateType.empty,
+        );
+}
+
+enum DataStateType { success, empty, error }
```
Comment (FranciscoGaraventa) [inline lib/src/core/resource/data_state.dart]: trailing comma

File: lib/src/data/datasource/remote/recipes_api_service.dart [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      } else {
+        return DataFailed(
+          '$textDataFailed ${response.statusCode}',
+        );
+      }
+    } catch (exception) {
+      return const DataFailed(
+        exceptionDataFailed,
+      );
+    }
```
Comment (FranciscoGaraventa) [inline lib/src/data/datasource/remote/recipes_api_service.dart]: Here you can use Dio package too, I suggested to use DioError to catch errors

File: lib/src/data/datasource/remote/recipes_api_service.dart [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  RecipesApiService();
+
+  static const String uri = 'https://api.spoonacular.com/';
+  static const String apiKey = '6add5cb5264c4077a6540c34ca241c8b';
+  static const String uriDio = '${uri}recipes/random?apiKey=$apiKey&number=3';
+  static const String textDataFailed =
+      'An error has occurred while loading recipes random';
+  static const String exceptionDataFailed = 'There was a request error';
+
+  Dio dio = Dio();
```
Comment (FranciscoGaraventa) [inline lib/src/data/datasource/remote/recipes_api_service.dart]: Inside Dio constructor you can set the baseUrl, in this case is 'uri'

File: lib/src/data/datasource/remote/recipes_api_service.dart [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import 'package:spoonacular/src/core/resource/data_state.dart';
+import 'package:spoonacular/src/data/model/recipes_list_model.dart';
+import 'package:spoonacular/src/domain/entity/recipes_list_entity.dart';
+
+class RecipesApiService {
+  RecipesApiService();
+
+  static const String uri = 'https://api.spoonacular.com/';
+  static const String apiKey = '6add5cb5264c4077a6540c34ca241c8b';
+  static const String uriDio = '${uri}recipes/random?apiKey=$apiKey&number=3';
```
Comment (FranciscoGaraventa) [inline lib/src/data/datasource/remote/recipes_api_service.dart]: You can use the apiKey as a query parameter, but you can also use it like this
`dio.get("endpoint", options: dio.Options(headers: headers))`
headers is a Map, inside this map you can set you apiKey

File: lib/src/presentation/controller/recipes_controller.dart [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import 'package:spoonacular/src/domain/entity/recipes_list_entity.dart';
+
+class RecipesController extends GetxController
+    with StateMixin<List<RecipeEntity>> {
+  RecipesController(this.recipesUseCase);
+
+  final UseCaseInterface recipesUseCase;
+
+  void getRecipesRandom() async {
+    change(null, status: RxStatus.loading());
```
Comment (FranciscoGaraventa) [inline lib/src/presentation/controller/recipes_controller.dart]: trailing comma

File: lib/src/presentation/view/home_page.dart [hunk 6]
```diff
@@ -10,20 +20,173 @@ class HomePage extends StatefulWidget {
...
+
+  final RecipesController controller = Get.put(
+    RecipesController(
+      RecipesUseCase(
+        RecipesRepository(
+          RecipesApiService(),
+        ),
+      ),
+    ),
+  );
```
Comment (FranciscoGaraventa) [inline lib/src/presentation/view/home_page.dart]: Instead to define all of this here, you can create a di field, in that file you are going to define, controllers, useCases, apiService, Dio, etc

File: lib/src/presentation/view/splash_screen.dart [hunk 7]
```diff
@@ -12,6 +10,11 @@ class SplashScreen extends StatefulWidget {
 }
 
 class _SplashScreenState extends State<SplashScreen> {
+  static const double splashScreenGifHeight = 0.7;
+  static const double splashScreenGifWidth = 0.5;
+  static const int splashScreenDuration = 3000;
+  static const String splashScreenGif = 'assets/lottie/prepare_food.json';
```
Comment (FranciscoGaraventa) [inline lib/src/presentation/view/splash_screen.dart]: Move all assets to a asset file

File: lib/src/presentation/widget/recipe_card.dart [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+    return [
+      _getTextWithStyle(recipe.vegetarian, recipeCardTextVegetarian),
+      _getTextWithStyle(recipe.vegan, recipeCardTextVegan),
+      _getTextWithStyle(recipe.glutenFree, recipeCardTextGlutenFree),
+      _getTextWithStyle(recipe.dairyFree, recipeCardTextDairyFree),
+      _getTextWithStyle(recipe.veryHealthy, recipeCardTextVeryHealthy),
+      _getTextWithStyle(recipe.cheap, recipeCardTextCheap),
+      _getTextWithStyle(recipe.veryPopular, recipeCardTextVeryPopular),
+      _getTextWithStyle(recipe.sustainable, recipeCardTextSustainable),
+      _getTextWithStyle(recipe.lowFodmap, recipeCardTextLowFodmap),
```
Comment (FranciscoGaraventa) [inline lib/src/presentation/widget/recipe_card.dart]: trailing comma

File: lib/src/presentation/widget/recipe_card.dart [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+                  imageUrl: recipe.image,
+                  imageBuilder: (context, imageProvider) => Container(
+                    decoration: BoxDecoration(
+                      image: DecorationImage(
+                        image: imageProvider,
+                        fit: BoxFit.fill,
+                      ),
+                    ),
+                  ),
+                  placeholder: (context, url) => Center(
```
Comment (FranciscoGaraventa) [inline lib/src/presentation/widget/recipe_card.dart]: trailing comma

File: lib/src/presentation/widget/recipe_card.dart [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+                    child: SizedBox(
+                      child: Lottie.asset(
+                        StringsConstants.imageLoadingUtensils,
+                      ),
+                    ),
+                  ),
+                  errorWidget: (context, url, error) => Center(
```
Comment (FranciscoGaraventa) [inline lib/src/presentation/widget/recipe_card.dart]: trailing comma

File: lib/src/core/resource/data_state.dart (hunk 1 above)
Comment (briansetzesglobant) [reply lib/src/core/resource/data_state.dart]: Done

File: lib/src/data/datasource/remote/recipes_api_service.dart (hunk 2 above)
Comment (briansetzesglobant) [reply lib/src/data/datasource/remote/recipes_api_service.dart]: Done

File: lib/src/data/datasource/remote/recipes_api_service.dart (hunk 3 above)
Comment (briansetzesglobant) [reply lib/src/data/datasource/remote/recipes_api_service.dart]: Done

File: lib/src/data/datasource/remote/recipes_api_service.dart (hunk 4 above)
Comment (briansetzesglobant) [reply lib/src/data/datasource/remote/recipes_api_service.dart]: Done

File: lib/src/presentation/controller/recipes_controller.dart (hunk 5 above)
Comment (briansetzesglobant) [reply lib/src/presentation/controller/recipes_controller.dart]: Done

File: lib/src/presentation/view/home_page.dart (hunk 6 above)
Comment (briansetzesglobant) [reply lib/src/presentation/view/home_page.dart]: Done

File: lib/src/presentation/view/splash_screen.dart (hunk 7 above)
Comment (briansetzesglobant) [reply lib/src/presentation/view/splash_screen.dart]: Done

File: lib/src/presentation/widget/recipe_card.dart (hunk 8 above)
Comment (briansetzesglobant) [reply lib/src/presentation/widget/recipe_card.dart]: Done

File: lib/src/presentation/widget/recipe_card.dart (hunk 9 above)
Comment (briansetzesglobant) [reply lib/src/presentation/widget/recipe_card.dart]: Done

File: lib/src/presentation/widget/recipe_card.dart (hunk 10 above)
Comment (briansetzesglobant) [reply lib/src/presentation/widget/recipe_card.dart]: Done

File: lib/src/data/repository/recipes_repository.dart [hunk 11]
```diff
@@ -0,0 +1,16 @@
+import 'package:get/get.dart';
+import 'package:spoonacular/src/core/resource/data_state.dart';
+import 'package:spoonacular/src/data/datasource/remote/recipes_api_service.dart';
+import 'package:spoonacular/src/domain/entity/recipes_list_entity.dart';
+import 'package:spoonacular/src/domain/repository/repository_interface.dart';
+
+class RecipesRepository extends RepositoryInterface {
+  //RecipesRepository(this.recipesApiService);
```
Comment (FranciscoGaraventa) [inline lib/src/data/repository/recipes_repository.dart]: Remove commented line

File: lib/src/domain/use_case/implementation/recipes_use_case.dart [hunk 12]
```diff
@@ -0,0 +1,17 @@
+import 'package:get/get.dart';
+import 'package:spoonacular/src/core/resource/data_state.dart';
+import 'package:spoonacular/src/core/use_case/use_case_interface.dart';
+import 'package:spoonacular/src/data/repository/recipes_repository.dart';
+import 'package:spoonacular/src/domain/entity/recipes_list_entity.dart';
+import 'package:spoonacular/src/domain/repository/repository_interface.dart';
+
+class RecipesUseCase extends UseCaseInterface<DataState<RecipesListEntity>> {
+  //RecipesUseCase(this.recipesRepository);
```
Comment (FranciscoGaraventa) [inline lib/src/domain/use_case/implementation/recipes_use_case.dart]: Remove commented line

File: lib/src/presentation/controller/recipes_controller.dart [hunk 13]
```diff
@@ -0,0 +1,41 @@
+import 'package:get/get.dart';
+import 'package:spoonacular/src/core/resource/data_state.dart';
+import 'package:spoonacular/src/core/use_case/use_case_interface.dart';
+import 'package:spoonacular/src/domain/entity/recipe_entity.dart';
+import 'package:spoonacular/src/domain/entity/recipes_list_entity.dart';
+import 'package:spoonacular/src/domain/use_case/implementation/recipes_use_case.dart';
+
+class RecipesController extends GetxController
+    with StateMixin<List<RecipeEntity>> {
+  //RecipesController(this.recipesUseCase);
```
Comment (FranciscoGaraventa) [inline lib/src/presentation/controller/recipes_controller.dart]: Remove commented line

File: lib/src/domain/use_case/implementation/recipes_use_case.dart (hunk 12 above)
Comment (briansetzesglobant) [reply lib/src/domain/use_case/implementation/recipes_use_case.dart]: done

File: lib/src/data/repository/recipes_repository.dart (hunk 11 above)
Comment (briansetzesglobant) [reply lib/src/data/repository/recipes_repository.dart]: done

File: lib/src/presentation/controller/recipes_controller.dart (hunk 13 above)
Comment (briansetzesglobant) [reply lib/src/presentation/controller/recipes_controller.dart]: done

**Review by FranciscoGaraventa — APPROVED** [review]

**Pull request merged**